Calibration/HcalCalibAlgos/src/AnalyzerMinbias.cc

```diff
@@ -0,0 +1,542 @@
+  edm::EDGetTokenT<HORecHitCollection>    tok_horecoMB_,   tok_horecoNoise_;
+  bool theRecalib_, ignoreL1_;
+  edm::EDGetTokenT<HBHERecHitCollection>  tok_hbheNormal_;
+  edm::EDGetTokenT<L1GlobalTriggerObjectMapRecord> tok_hltL1GtMap_;
+};
+
+AnalyzerMinbias::AnalyzerMinbias(const edm::ParameterSet& iConfig, 
+				 const HcalMinbias::Counters* counters) {
+
+  // get name of output file with histogramms
+  
+  
+  // get token names of modules, producing object collections
+  tok_hbherecoMB_   = consumes<HBHERecHitCollection>(iConfig.getParameter<edm::InputTag>("hbheInputMB"));
+  tok_horecoMB_     = consumes<HORecHitCollection>(iConfig.getParameter<edm::InputTag>("hoInputMB"));
+  tok_hfrecoMB_     = consumes<HFRecHitCollection>(iConfig.getParameter<edm::InputTag>("hfInputMB"));
+
+  tok_hbherecoNoise_= consumes<HBHERecHitCollection>(iConfig.getParameter<edm::InputTag>("hbheInputNoise"));
+  tok_horecoNoise_  = consumes<HORecHitCollection>(iConfig.getParameter<edm::InputTag>("hoInputNoise"));
+  tok_hfrecoNoise_  = consumes<HFRecHitCollection>(iConfig.getParameter<edm::InputTag>("hfInputNoise"));
+
+  theRecalib_       = iConfig.getParameter<bool>("Recalib"); 
+  ignoreL1_         = iConfig.getUntrackedParameter<bool>("IgnoreL1", true);
+  runNZS_           = iConfig.getUntrackedParameter<bool>("RunNZS", true);
+
+
+  tok_hbheNormal_   = consumes<HBHERecHitCollection>(edm::InputTag("hbhereco"));
+  tok_hltL1GtMap_   = consumes<L1GlobalTriggerObjectMapRecord>(edm::InputTag("hltL1GtObjectMap"));
+}
+  
+AnalyzerMinbias::~AnalyzerMinbias() { }
+  
+void AnalyzerMinbias::endStream() {
+
+  for (std::map<std::pair<int,HcalDetId>,HcalMinbias::myInfo>::const_iterator itr=myMap_.begin(); itr != myMap_.end(); ++itr) {
+    std::pair<int,HcalDetId> id = itr->first;
+    std::map<std::pair<int,HcalDetId>,HcalMinbias::myInfo>::iterator itr1 = (globalCache()->myMap_).find(id);
+    if (itr1 == globalCache()->myMap_.end()) {
+      HcalMinbias::myInfo info;
+      globalCache()->myMap_[id] = info;
+      itr1 = (globalCache()->myMap_).find(id);
+      itr1->second.runcheck = itr->second.runcheck;
+    }
+    itr1->second.theNS0  += itr->second.theNS0;
+    itr1->second.theNS1  += itr->second.theNS1;
+    itr1->second.theNS2  += itr->second.theNS2;
+    itr1->second.theNS3  += itr->second.theNS3;
+    itr1->second.theNS4  += itr->second.theNS4;
+    itr1->second.theMB0  += itr->second.theMB0;
+    itr1->second.theMB1  += itr->second.theMB1;
+    itr1->second.theMB2  += itr->second.theMB2;
+    itr1->second.theMB3  += itr->second.theMB3;
+    itr1->second.theMB4  += itr->second.theMB4;
+    itr1->second.theDif0 += itr->second.theDif0;
+    itr1->second.theDif1 += itr->second.theDif1;
+    itr1->second.theDif2 += itr->second.theDif2;
+  }
+}
+  
+//  EndJob
+//
+void AnalyzerMinbias::globalEndJob(const HcalMinbias::Counters* count) {
+   
+  TFile* hOutputFile_ = new TFile(count->fOutputFileName_.c_str(), "RECREATE") ;
```

Comment:
This is a memory leak. There is no need to use ```new``` here so instead I suggest just make the ```TFile``` on the stack so it automatically gets cleaned up.